visidata/guide.py

```diff
@@ -156,15 +160,27 @@ class GuideSheet(Sheet):
             ItemColumn('guide', 1, width=80, displayer='full'),
             ]
     precious = False
-    guide_text = ''
-    sheettype = Sheet
+    sheettype = 'Sheet'
 
     def iterload(self):
+        text = self.source.open(mode='r').read()
         winWidth = 78
-        helper = AttrDict(commands=CommandHelpGetter(self.sheettype),
+        helper = AttrDict(commands=CommandHelpGetter(globals()[self.sheettype]),
                           options=OptionHelpGetter())
-        guidetext = MissingAttrFormatter().format(self.guide_text, help=helper, vd=vd)
-        for startingLine, text in enumerate(guidetext.splitlines()):
+        guidetext = MissingAttrFormatter().format(text, help=helper, vd=vd)
+
+        # parsing front matter
+        sections = guidetext.split('---\n', maxsplit=2)
+        d = {}
+        for config in sections[:-1]:
+            config = config.strip()
+            if config:
+                key, val = config.split(': ', maxsplit=1)
+                d[key] = val
+        self.__dict__.update(d)
```

Comment:
What might this be used for, besides `sheettype`?

Reply:
I had a vision for `VisiData version`, `updated on`. I also thought it might be good to add this mechanism in case we do add to these parameters. Maybe we designate categories of guides, where only some categories get concatenated into a manpage later.

Reply:
Well I'm wondering if we shouldn't store it as an AttrDict `metadata` or something, and then it would be clearer at the place where it's used, what's going on and where it comes from.

Reply:
Great idea! I implemented that.